### PR TITLE
Cleanup: add docker build to README and CONTRIBUTING to cover godep

### PR DESCRIPTION
**Pull request by chanwit**
`Godep` kicks in. This PR updates the document to cover how to compile `swarm` with it.

Signed-off-by: Chanwit Kaewkasi <chanwit@gmail.com>

### CODE REVIEW

**Comment by bfirsh** [comment]
I'd much prefer to just recommend:

```
$ docker build -t swarm .
$ docker run swarm ...
```

**Comment by nathanleclaire** [comment]
+1 @bfirsh suggested direction.  The `docker-machine` build happens in a container and the binaries end up on the host via bind mount, this has been working out really well for us and makes it quite simple to get started developing `docker-machine` on a new computer.

**Comment by chanwit** [comment]
I am not working only amd64. Basically, `docker build` is not my option to port `swarm` to other platforms.

**Comment by aluzzardi** [comment]
I believe we need to overhaul the README to also take into account that we're going to have an "official" image soon.

Which means that the first paragraph on getting started should be something like `docker run swarm`, as most people don't care about building and compiling from source.

I agree we have to talk about godep - I'm wondering if it should be in the README or CONTRIBUTING, since those steps are meant for developers?

/cc @SvenDowideit

**Comment by nathanleclaire** [comment]
> I am not working only on amd64. Basically, docker build is not my option to port swarm to other platforms.

`docker-machine` uses [gox](https://github.com/mitchellh/gox) for this (it will spit out binaries for all platforms and arches supported by Go by default IIRC - on `docker-machine` it is set to default to all possible combinations of Linux/Windows/OSX and amd64/i386).

It is also possible to specify manually using `GOOS` or `GOARCH` environment variables if needed.  If the environment is set up correctly and cgo dependencies are not used cross-compilation shouldn't be an issue.

**Comment by chanwit** [comment]
@nathanleclaire Thank you for pointing this out.

@aluzzardi Yep, I agreed that `godep` related documents would be in CONTRIBUTING.

So what suggested by @bfirsh and @nathanleclaire should be put into README. But it seems to not be ready yet.

Shall we remove the installation paragraph out of README at the moment and wait until the build process via `docker` available then?

**Comment by chanwit** [comment]
After reading the new Dockerfile, I tweaked README to add `docker build`.

CONTRIBUTING may still need to cover how to `godep save` and `godep restore` properly when we want to add new dependencies.

**Comment by chanwit** [comment]
ping @aluzzardi @vieux

**Comment by vieux** [comment]
@chanwit we are waiting to have the official image to integrate your changes in a `Contributing.md` document

**Comment by chanwit** [comment]
@vieux thank you for letting me know. I'll try to keep up with those changes to make the doc valid.

**Comment by vieux** [comment]
@chanwit can you please rebase

**Comment by chanwit** [comment]
@vieux thanks for the ping. I've rebased and finaized things.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,4 +10,46 @@ Docker does things, you'll feel right at home.
...
+
+Then you can fetch `swarm` code to your `$GOPATH`.
+```sh
+go get -d github.com/docker/swarm
+```
+
+The final step is to prepare dependencies for `swarm` and compile it.
+```sh
+cd src/github.com/docker/swarm
+$GOBIN/godep restore
```
Comment (bacongobbler) [inline CONTRIBUTING.md]: FYI instead of restoring, `$GOBIN/godep go install` will use Godep's workspace without cluttering up your $GOPATH

Comment (chanwit) [inline CONTRIBUTING.md]: Thank you for the comment @bacongobbler. I am aware of that.

However, `godep restore` is a proper workflow when do coding. Then after we finishing the code, we do `go dep save` and try compliing it with `godep go install` before pushing code upstream. I should clarify steps in the workflow any way.

**Comment by chanwit** [comment]
@aluzzardi @vieux could you please take a look.
Thank you.

**Comment by vieux** [comment]
@aluzzardi @SvenDowideit can you take a look at this ?

**Comment by aluzzardi** [comment]
ping @SvenDowideit

**Comment by vieux** [comment]
@SvenDowideit can you take a look ?

**Comment by SvenDowideit** [comment]
LGTM - sorry @chanwit a rebase will be needed.

**Comment by vieux** [comment]
ping @chanwit

**Comment by chanwit** [comment]
Sorry for being late @SvenDowideit @vieux. The doc has been rebased.

**Comment by vieux** [comment]
LGTM

**Pull request merged**